src/solvers/rosenbrock.jl

```diff
@@ -1,93 +1,239 @@
 export SSPKnoth
+using StaticArrays
+import DiffEqBase
+import LinearAlgebra: ldiv!, diagm
+import LinearAlgebra
 
-abstract type RosenbrockAlgorithm <: DistributedODEAlgorithm end
+abstract type RosenbrockAlgorithmName <: AbstractAlgorithmName end
 
-struct RosenbrockTableau{N, RT, N²}
-    A::SMatrix{N, N, RT, N²}
-    C::SMatrix{N, N, RT, N²}
-    Γ::SMatrix{N, N, RT, N²}
-    m::SMatrix{N, 1, RT, N}
+"""
+    RosenbrockTableau{N, RT, N²}
+
+Contains everything that defines a Rosenbrock-type method.
+
+- N: number of stages,
+
+Refer to the documentation for the precise meaning of the symbols below.
+"""
+struct RosenbrockTableau{N}
```

Comment:
I'd like to keep the tableau structs confined to the minimal set of values needed to define a timestepping scheme, and to have all values generated from those confined to the cache. The `ExplicitTableau` (which I'll rename to `ButcherTableau` in the reformulation) has the vector `b`, which you defined elsewhere in this file, and the matrix `A`, which corresponds to your matrices `α` and `Γ`. The `ExplicitTableau` also has the vector `c`, which I think corresponds to the values `αi` and `γi` that you compute in your stage loop; it might be worth pre-computing those here as well. So, maybe the constructor here can look something like `RosenbrockTableau(α, Γ, b, c = sum(α, dims = 2), γ = sum(Γ, dims = 2))`?

Reply:
I added constructor that takes `alpha`, `Gamma`, and `b`. Maybe we can refine things once we work with the reformulation?